OpenSim/Common/DelimFileAdapter.h

```diff
@@ -22,19 +22,72 @@
 #ifndef OPENSIM_DELIM_FILE_ADAPTER_H_
 #define OPENSIM_DELIM_FILE_ADAPTER_H_
 
+#include "SimTKcommon.h"
+
 #include "FileAdapter.h"
 #include "TimeSeriesTable.h"
 
+#include <string>
+#include <fstream>
+#include <regex>
+
 namespace OpenSim {
 
+class IncorrectNumTokens : public Exception {
+public:
+    IncorrectNumTokens(const std::string& file,
+                       size_t line,
+                       const std::string& func,
+                       const std::string& msg) :
+        Exception(file, line, func) {
+
+        addMessage(msg);
+    }
+};
+
+class DataTypeMismatch : public Exception {
+public:
+    DataTypeMismatch(const std::string& file,
+                     size_t line,
+                     const std::string& func,
+                     const std::string& expected,
+                     const std::string& received) :
+        Exception(file, line, func) {
+        std::string msg = "expected = " + expected;
+        msg += " received = " + received;
+
+        addMessage(msg);
+    }
+};
+
+namespace {
+    template<typename T>
+    struct is_SimTK_Vec : std::false_type {};
+
+    template<int M, typename ELT, int Stride>
+    struct is_SimTK_Vec<SimTK::Vec<M, ELT, Stride>> {
+        static constexpr bool value = (M >= 2 && M <= 12);
+    };
+} // namespace
+
 /** DelimFileAdapter is a FileAdapter that reads and writes text files with
 given delimiters. CSVFileAdapter and MOTFileAdapter derive from this class and
 set the delimiters appropriately for the files they parse. The read/write
 functions return/accept a specific type of DataTable referred to as Table in 
 this class.                                                                   
 Header in the file is assumed to end with string "endheader" occupying a full
 line.                                                                         */
+template<typename T>
 class OSIMCOMMON_API DelimFileAdapter : public FileAdapter {
```

Comment:
Since the `DelimFileAdapter` is in header defined there is no need for the dllimport/export decls defined in `OSIMCOMMON_API`. In fact, this may be causing the AppVeyor [failure](https://ci.appveyor.com/project/opensim-org/opensim-core/build/4547/job/ef5e3h7aiu3sr09t#L303) since the complete definition is in this module when it is expecting it elsewhere.